test/RequestChannelTest.cpp

```diff
@@ -88,11 +88,299 @@ TEST(RequestChannelTest, RequestOnDisconnectedClient) {
   ASSERT(did_call_on_error);
 }
 
-// TODO complete from requester, responder continues
-// TODO complete from responder, requester continues
-// TODO cancel from requester, shuts down
-// TODO flow control from requester to responder
-// TODO flow control from responder to requester
-// TODO failure on responder, requester sees
-// TODO failure on request, requester sees
-// TODO failure from requester ... what happens?
+class TestChannelResponder : public rsocket::RSocketResponder {
+ public:
+  TestChannelResponder(
+      int64_t rangeEnd = 10,
+      int64_t initialSubReq = credits::kNoFlowControl)
+      : rangeEnd_{rangeEnd},
+        testSubscriber_{TestSubscriber<std::string>::create(initialSubReq)} {}
+
+  yarpl::Reference<Flowable<rsocket::Payload>> handleRequestChannel(
+      rsocket::Payload initialPayload,
+      yarpl::Reference<Flowable<rsocket::Payload>> requestStream,
+      rsocket::StreamId) override {
+    // add initial payload to testSubscriber values list
+    testSubscriber_->onNext(initialPayload.moveDataToString());
```

Comment:
I guess the testSubscriber doesn't mind you don't call onSubscribe before onNext?

Reply:
All it does is add it to it's internal vector of values.  Adding the initial payload let me stream the same things both directions and then make sure they were the same.  It's not truly necessary.